numpy/lib/twodim_base.py

```diff
@@ -32,6 +32,76 @@ def _min_int(low, high):
     return int64
 
 
+def flip(m, axis):
+    """
+    Reverse the order of elements in an array along the given axis.
+
+    All other axes are preserved, but appear in a different order than before.
+
+    Parameters
+    ----------
+    m : array_like
+        Input array.
+    axis: integer
+        Axis in array, which entries are reversed.
+        
+
+    Returns
+    -------
+    out : array_like
+        A view of `m` with the entries of axis reversed.  Since a view is
+        returned, this operation is :math:`\\mathcal O(1)`.
+
+    See Also
+    --------
+    flipud : Flip an array vertically (axis=0).
+    fliplr : Flip an array horizontally (axis=1).
+
+    Notes
+    -----
+    flip(m, 0) is equivalent to flipud(m).
+    flip(m, 1) is equivalent to fliplr(m).
+    flip(m, n) corresponds to ``A[...,::-1,...]`` with ``::-1`` at position n.
+
+    Examples
+    --------
+    >>> A = np.arange(8).reshape((2,2,2))
+    >>> A
+    array([[[0, 1],
+            [2, 3]],
+
+           [[4, 5],
+            [6, 7]]])
+
+    >>> flip(A, 0)
+    array([[[4, 5],
+            [6, 7]],
+    
+           [[0, 1],
+            [2, 3]]])
+
+    >>> flip(A, 1)
+    array([[[2, 3],
+            [0, 1]],
+    
+           [[6, 7],
+            [4, 5]]])
+
+    >>> A = np.random.randn(3,4,5)
+    >>> np.all(flip(A,2)==A[:,:,::-1,...])
+    True
+    """
+    if axis==0:
```

Comment:
In general, clarity is more important than unmeasured micro-optimizations.

On Fri, Feb 26, 2016 at 3:10 PM, C. Eren Sezener
<notifications@github.com> wrote:
> In numpy/lib/twodim_base.py:
>
>> +
>> +           [[0, 1],
>> +            [2, 3]]])
>> +
>> +    >>> flip(A, 1)
>> +    array([[[2, 3],
>> +            [0, 1]],
>> +
>> +           [[6, 7],
>> +            [4, 5]]])
>> +
>> +    >>> A = np.random.randn(3,4,5)
>> +    >>> np.all(flip(A,2)==A[:,:,::-1,...])
>> +    True
>> +    """
>> +    if axis==0:
>
> They are not necessary but flipud and fliplr should be slightly faster.
> Should we remove the cases?
>
> —
> Reply to this email directly or view it on GitHub.



-- 
Nathaniel J. Smith -- https://vorpus.org